cg/cli/deliver/utils.py

```diff
@@ -43,3 +44,11 @@ def deliver_raw_data_for_analyses(
             )
             LOG.error(f"Could not deliver files for analysis {analysis.id}: {error}")
             continue
+
+
+def get_pseudo_workflow(case: Case) -> str:
+    """Return the literal '' if the case is Microbial-fastq, otherwise return the case workflow."""
+    tag: str = case.samples[0].application_version.application.tag
+    if case.data_analysis == Workflow.RAW_DATA and tag in MICROBIAL_APP_TAGS:
+        return "microbial-fastq"
+    return case.data_analysis
```

Comment:
add this function to the factory class in: cg/services/deliver_files/deliver_files_service/deliver_files_service_factory.py

```suggestion
def convert_workflow(case: Case) -> Workflow:
    """Converts a workflow.With the introduction of the microbial-fastq delivery type an unsupported combination of delivery type and workflow setup is required. This function makes sure that a raw data workflow with microbial fastq delivery type is treated as a microsalt workflow so that the microbial-fastq sample files can be concatenated."""
    tag: str = case.samples[0].application_version.application.tag
    if case.data_analysis == Workflow.RAW_DATA and case.delivery_type == "microbial-fastq"
        return Workflow.MICROSALT
    return case.data_analysis
```
 add this line to the function build_delivery_service in cg/services/deliver_files/deliver_files_service/deliver_files_service_factory.py

```suggestion
workflow: Workflow = self.convert_workflow(workflow,delivery_type)
```

Reasoning: 

We also use this factory in `cg/meta/upload/upload_api.py` with the current implementation we would need to also fix the workflow fetching in that place. 

On the other hand, if we let the factory handle the conversion other places in the code do not need to know about that a conversion is happening. I think this would lead to the least changes overall and the logic is kept within the delivery module.

Reply:
The problem I see here is that "microbial-fastq" is not a case DataDelivery option, as these refer to Scout, Analysis-fastq etc. We would then still need to test the tag

Reply:
Yes, you are right! The tags would still be required then!

Reply:
implemented in factory class